compiler/rustc_ty_utils/src/instance.rs

```diff
@@ -53,6 +53,15 @@ fn resolve_instance<'tcx>(
                     _ => return Ok(None),
                 }
 
+                // FIXME(#127030): `ConstParamHasTy` has bad interactions with
```

Comment:
We could actually move this change to:

https://github.com/rust-lang/rust/blob/9c3bc805dd9cb84019c124b9a50fdff1e62a7ec9/compiler/rustc_mir_transform/src/inline.rs#L285

Reply:
But I am kinda worried that there are other calls to `Instance::resolve` that may trigger this problem. Not exactly sure why -- I could audit all of the `Instance::resolve` calls in the codebase, I guess.